src/vmm/src/gdb/kvm_debug.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Modified functions from `kvm-ioctls` to take `RawFd` instead of a `VcpuFd`
+use std::os::fd::RawFd;
+
+use kvm_bindings::*;
+use kvm_ioctls::kvm_ioctls::*;
+use vm_memory::GuestAddress;
+use vmm_sys_util::errno;
+use vmm_sys_util::ioctl::{ioctl_with_mut_ref, ioctl_with_ref};
+
+use super::target::Error;
+
+/// Sets the 9th (Global Exact Breakpoint enable) and the 10th (always 1) bits for the DR7 debug
+/// control register
+const X86_GLOBAL_DEBUG_ENABLE: u64 = 0b11 << 9;
+
+/// Retrieves the last error and stores it in a target Error
+fn last_error() -> Error {
+    Error::KvmIoctlsError(errno::Error::last())
+}
+
+/// Returns the vCPU general purpose registers.
+///
+/// See: https://github.com/rust-vmm/kvm-ioctls/blob/ae039f245db0cd40ac20c32049ec536222b6dea7/src/ioctls/vcpu.rs#L226
+#[cfg(target_arch = "x86_64")]
+pub fn get_regs(kvm_fd: &RawFd) -> Result<kvm_regs, Error> {
```

Comment:
Aren't all of these methods work with `vcpu` fd?

Reply:
Ah yeah I forgot to update these names I'll do that